File: Dataset/Leetcode/valid/101/514.py
 class Solution:
    def XXX(self, root: TreeNode) -> bool:
        if not root:
            return True
        def lr(r1,r2):
            if not r1 and not r2:
                return True
            if not r1 or not r2:
                return False
            if r1.val !=r2.val:
                return False
            return lr(r1.left,r2.right) and lr(r1.right,r2.left)
        return lr(root.left,root.right)

File: Dataset/Leetcode/train/94/514.cpp
 class Solution {
public:
    void dfs(TreeNode *node, vector<int> & ans)
    {
        if (!node) return;
        dfs(node->left, ans);
        ans.push_back(node->val);
        dfs(node->right, ans);
    }
    vector<int> XXX(TreeNode* root) 
    {
        vector<int> ans;
        dfs(root, ans);
        return ans;
    }
};

